CHANGELOG.md

```diff
@@ -2,6 +2,36 @@
 
 Changes. Changes everywhere!
 
+## 1.0.0-beta.12
+###### _TBA_
+
+### Breaking change
+
+- [typescript] Improve type definition for withStyles (#8320)
+
+@pelotom did a great job improving the `withStyles` typings, such that less generics are required to be written! Most notably, you do no longer have to pass a map of class names to `withStyles`:
+
+```diff
+- withStyles<{ root: string; }>(...)
++ withStyles<'root'>(...)
+```
+
+Also, `props` can now be set when applying the HOC:
+
+```diff
+- const StyledComponent = withStyles<
+-   StyledComponentProps,
+-   StyledComponentClassNames
+- >(styles)(Component);
++ const StyledComponent = withStyles(styles)<StyledComponentProps>(
++   ({ classes, text }) => (
++     <div className={classes.root}>
++       {text}
++     </div>
++   )
++ );
+```
+
```

Comment:
You might want to also mention that 
- you no longer need to pass a dummy `classes` argument to decorated components (#8267), but that
- using `withStyles` as a decorator in TypeScript doesn't work in a well-typed way due to [TypeScript #4881](https://github.com/Microsoft/TypeScript/issues/4881). To "decorate" a class, instead of

```ts
@withStyles(styles)
class DecoratedComponent extends React.Component<
  Props & { classes: { root: string } }
> {
  render() {
    const { classes, text } = this.props;
    return (
      <div className={classes.root}>
        {text}
      </div>
    );
  }
}
```

you need to write something like

```ts
const DecoratedComponent = withStyles(styles)(
  class extends React.Component<WithStyles<Props, 'root'>> {
    render() {
      return (
        <div className={this.props.classes.root}>
          {this.props.text}
        </div>
      );
    }
  }
)
```

Reply:
Ok, if #8354 is merged we can strike the above and instead just mention that
- when `withStyles` is used as a decorator and `strictNullChecks` is enabled, one has to use the `!` operator to access `classes` from within the class, and
- #8267 is fixed